src/Servers/Kestrel/Core/src/Internal/BufferWriter.cs

```diff
@@ -69,6 +69,9 @@ public void Commit()
                 _bytesCommitted += buffered;
                 _buffered = 0;
                 _output.Advance(buffered);
+                
+                // Calling IBufferWriter<T>.Advance(int) may have invalidated our cached Span<T>, so reacquire next time.
+                _span = default;
```

Comment:
This will regress `EnsureMore` which immediately sets the span after `Commit`, so should be factored differently